aten/src/ATen/native/native_functions.yaml

```diff
@@ -29,6 +29,12 @@
 - func: _cast_Half(Tensor self, bool non_blocking=False) -> Tensor
   variants: function
 
+- func: backward(Tensor(a!) self, Tensor? gradient=None, bool keep_graph=False, bool create_graph=False) -> void
```

Comment:
how did you choose these annotations?  For example, why is backward's `self` marked as writeable, but set_data isn't?

Reply:
I'm not even really sure what these annotations mean in this context, as they are pretty different than the usual meanings.

For example, `backward` is the only place where we mutate `self.grad` -- does that count as mutable?  `self` itself doesn't actually change.

Maybe @suo can give some guidance here.

Reply:
`backward()` is a special case that isn't expressible in the alias annotation language—it writes to every tensor in the autograd graph as the gradients are accumulated to the leaves. So if we enabled in the JIT it would have to be a special case in the alias analysis pass.

The bit about mutating the `grad` property only isn't that relevant—for the purpose of optimization, you can't reorder a call to `backward()` across a read of any tensor in the autograd graph anyway, so it's fine to consider `self.grad` part of the `self`'s memory.